rclpy/test/test_executor.py

```diff
@@ -302,6 +302,47 @@ def test_executor_add_node(self):
         assert not executor.add_node(self.node)
         assert id(executor) == id(self.node.executor)
 
+    def test_executor_spin_until_future_complete(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor(context=self.context)
+        executor.add_node(self.node)
+
+        def timer_callback():
+            pass
+        timer = self.node.create_timer(0.1, timer_callback)
```

Comment:
Why do we need to create a timer? Is this so that the underlying `spin_once` will return (ie not timeout) and we can check the future? Seems like the spin_until_future_complete logic could be improved by not making it rely on other callbacks. This type of improvement does not have to happen in this PR since it looks like it has always been this way.

Reply:
Right, spin_until_future_complete has underlying `spin_once` like [rclcpp](https://github.com/ros2/rclcpp/blob/83beaf8a3f9c6ac417fcb170b24712f95c67c5c2/rclcpp/include/rclcpp/executor.hpp#L247).